pkg/tfgen/generate.go

```diff
@@ -858,6 +863,11 @@ func (g *Generator) Generate() error {
 		return errors.Wrapf(err, "failed to create Pulumi schema")
 	}
 
+	// Apply schema post-processing if defined in the provider.
```

Comment:
Depending on the surgery that's done in the callback, this may fail nameCheck pass below.

Reply:
Perhaps move the callback application after the namecheck pass?

Reply:
I'm a bit out of my depth here. It sounds fine to nameCheck after the surgery as an extra consistency check?

Reply:
nameCheck takes a 	spec pschema.PackageSpec and it's going to check  spec.Types. If you move around spec.Types, it will start complaining possibly... You may deal with it later as you hit the problem, I'd suggest just running namecheck on pre-modified PackageSpec.

Reply:
By "to the end", I mean post `g.UnstableGenerateFromSchema`